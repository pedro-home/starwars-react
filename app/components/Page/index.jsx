import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

const Component = ({ name, children }) => {
  return <div className={classnames('page', name)}>{children}</div>;
};

Component.propTypes = {
  name: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
};

export default Component;
