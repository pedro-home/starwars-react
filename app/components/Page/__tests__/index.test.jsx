import React from 'react';
import { render } from '@testing-library/react';

import Page from '../index';

describe('Given <Page />', () => {
  let component;
  let defaultProps;
  let text;
  let name;

  beforeAll(() => {
    text = 'Hello';
    name = 'TestPage';
    defaultProps = { children: <span>{text}</span>, name };
  });

  describe('When initializes', () => {
    beforeEach(() => {
      const { container } = render(<Page {...defaultProps} />);
      component = container;
    });

    it('Then should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('And should render children', () => {
      expect(component.querySelector('span').innerHTML).toEqual(text);
    });

    it('And should render class name with name', () => {
      expect(component.querySelector('.page').classList).toContainEqual(name);
    });
  });
});
