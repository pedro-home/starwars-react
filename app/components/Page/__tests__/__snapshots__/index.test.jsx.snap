// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <Page /> When initializes Then should match the snapshot 1`] = `
<div>
  <div
    class="page TestPage"
  >
    <span>
      Hello
    </span>
  </div>
</div>
`;
